code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -78,6 +78,8 @@
 		for(var/atom/T in get_turf(D))
 			D.reagents.reaction(T)
 		sleep(3)
+		if(QDELETED(D))
+			return
```

Comment:
I suspect just doing a continue in the `for(var/atom/T in get_turf(D))` where `T == D` will rectify this issue.

Reply:
I've looked at all the reactions possible and found no possible reactions that would remove a `/obj/effect/decal/chempuff`. `acid_act` is overridden at `/obj/effect/acid_act` to do nothing and `water_act` does nothing other than extinguish and set the `acid_level` to 0 (`/obj/water_act`).

So my conclusion would be that the effect got destroyed by other means. Like by calling `ex_act` while it sleeps on line 80.